tests/engine/adhoc_test.go

```diff
@@ -221,3 +225,41 @@ func importedAndExportedFunc(t *testing.T, newEngine func() wasm.Engine) {
 	require.NoError(t, err)
 	require.Equal(t, uint64(42), results[0])
 }
+
+// Test that host functions can handle float parameter without corrupting the value
+func hostFuncWithFloat32Param(t *testing.T, newEngine func() wasm.Engine) {
+	ctx := context.Background()
+	mod, err := text.DecodeModule([]byte(`(module
+		;; 'test_f32param' is a host function which accepts a f32 param
+		;; and tests if it is an expected value.
+		(import "env" "test_f32param" (func $test_f32param (param f32)))
+
+		;; 'call_test_f32' is an exported guest function to call
+		;; the host function, 'test_f32param', from the guest.
+		(func $call_test_f32param (param f32)
```

Comment:
the ID namespace is amazingly open so you can be very literal.

```suggestion
		;; call->test.f32param proxies test.f32param via call in order to test floats aren't corrupted.
		(func $call->test.f32param (param f32)
```